src/main/kotlin/leetcode/q1184/Q1184.kt

```diff
@@ -0,0 +1,26 @@
+package leetcode.q1184
+
+class Q1184 {
+    fun distanceBetweenBusStops(distance: IntArray, start: Int, destination: Int): Int {
+        val stops = distance.size
+
+        return if (start == destination) {
+            0
+        } else {
+            val min = min(start, destination)
+            val max = max(start, destination)
+            val sumBeforeMin = addBetween(distance, 0, min)
+            val sumBetweenMinAndMax = addBetween(distance, min, max)
+            val sumAfterMax = addBetween(distance, max, stops)
+            min(sumBeforeMin + sumAfterMax, sumBetweenMinAndMax)
+        }
+    }
+
+    private fun addBetween(distance: IntArray, start: Int, end: Int): Int {
+        var count = 0
+        for (i in start until end) {
+            count += distance[i]
+        }
+        return count
```

Comment:
return (start until end).reduce(0) {acc,i -> acc + distance[i]}

Reply:
@Castor-MS @rhkddud3917 

https://blog.leocat.kr/notes/2020/03/09/kotlin-reduce-and-fold